diff --git a/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte b/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
index 46df8b8..de17315 100644
--- a/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
+++ b/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
@@ -118,9 +118,11 @@
     dispatch("change", value);
   });
 
+  let inputValue = value;
+
   $: incrementLabel = translateWithId("increment");
   $: decrementLabel = translateWithId("decrement");
-  $: value = Number(value);
+  $: value = Number(inputValue);
   $: error =
     invalid || (!allowEmpty && value === "") || value > max || value < min;
   $: errorId = `error-${id}`;
@@ -181,7 +183,7 @@
           aria-label="{label ? undefined : ariaLabel}"
           on:input
           on:input="{({ target }) => {
-            value = target.value;
+            inputValue = target.value;
           }}"
           disabled="{disabled}"
           id="{id}"
@@ -230,7 +232,7 @@
           aria-label="{label ? undefined : ariaLabel}"
           on:input
           on:input="{({ target }) => {
-            value = target.value;
+            inputValue = target.value;
           }}"
           disabled="{disabled}"
           id="{id}"
