diff --git a/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte b/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
index 46df8b8..5a824a3 100644
--- a/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
+++ b/node_modules/carbon-components-svelte/src/NumberInput/NumberInput.svelte
@@ -118,9 +118,11 @@
     dispatch("change", value);
   });
 
+  let inputValue = value;
+
   $: incrementLabel = translateWithId("increment");
   $: decrementLabel = translateWithId("decrement");
-  $: value = Number(value);
+  $: value = Number(inputValue);
   $: error =
     invalid || (!allowEmpty && value === "") || value > max || value < min;
   $: errorId = `error-${id}`;
@@ -230,14 +232,14 @@
           aria-label="{label ? undefined : ariaLabel}"
           on:input
           on:input="{({ target }) => {
-            value = target.value;
+            inputValue = target.value;
           }}"
           disabled="{disabled}"
           id="{id}"
           max="{max}"
           min="{min}"
           step="{step}"
-          value="{value}"
+          value="{inputValue}"
           readonly="{readonly}"
         />
         {#if invalid}
